File: pfeBack/src/main/java/com/example/pfeBack/model/Message.java
package com.example.pfeBack.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;

@Document(collection = "messages")
public class Message {
    @Id
    private String id;
    private String senderId;
    private String receiverId;
    private String chatGroupId;
    private String content;
    private LocalDateTime timestamp;
    private String senderName;

    public Message() {}

    public Message(String senderId, String chatGroupId, String content, String senderName) {
        this.senderId = senderId;
        this.chatGroupId = chatGroupId;
        this.content = content;
        this.timestamp = LocalDateTime.now();
        this.senderName = senderName;
    }

    // Getters and Setters
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getSenderId() {
        return senderId;
    }

    public void setSenderId(String senderId) {
        this.senderId = senderId;
    }

    public String getReceiverId() {
        return receiverId;
    }

    public void setReceiverId(String receiverId) {
        this.receiverId = receiverId;
    }

    public String getChatGroupId() {
        return chatGroupId;
    }

    public void setChatGroupId(String chatGroupId) {
        this.chatGroupId = chatGroupId;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    public String getSenderName() {
        return senderName;
    }

    public void setSenderName(String senderName) {
        this.senderName = senderName;
    }
} 

File: Perform-up-App/lib/models/message.dart
class Message {
  final String id;
  final String senderId;
  final String receiverId;
  final String chatGroupId;
  final String content;
  final DateTime timestamp;
  final String senderName;

  Message({
    required this.id,
    required this.senderId,
    required this.receiverId,
    required this.chatGroupId,
    required this.content,
    required this.timestamp,
    required this.senderName,
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['id']?.toString() ?? '',
      senderId: json['senderId']?.toString() ?? '',
      receiverId: json['receiverId']?.toString() ?? '',
      chatGroupId: json['chatGroupId']?.toString() ?? '',
      content: json['content']?.toString() ?? '',
      timestamp: json['timestamp'] != null 
          ? DateTime.parse(json['timestamp'].toString()) 
          : DateTime.now(),
      senderName: json['senderName']?.toString() ?? 'Unknown',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'senderId': senderId,
      'receiverId': receiverId,
      'chatGroupId': chatGroupId,
      'content': content,
      'timestamp': timestamp.toIso8601String(),
      'senderName': senderName,
    };
  }
} 